internal/static/metrics/instance/instance_test.go

```diff
@@ -278,7 +278,6 @@ func TestInstance_Recreate(t *testing.T) {
 
 	// Recreate the instance, no panic should happen.
 	require.NotPanics(t, func() {
-		inst, err := New(prometheus.NewRegistry(), cfg, walDir, logger)
```

Comment:
This actually recreates a blank new Prom registry and the children write handler won't have any duplicate registration, thus no panic.

Removing it allows for reproduction of the panic

Reply:
I don't think we should remove this line (the comment doesn't recreate the instance anymore), but rather pass in the same registry on both line 261 and here. Can we add it back and make that change?

Reply:
Re-added :D

Reply:
And I can confirm that without the change in `instance.go`, that also reproduces the panic.